lib/client.js

```diff
@@ -1490,8 +1493,11 @@ function writeStream ({ client, request, socket, contentLength, header, expectsP
       }
 
       bytesWritten += len
+      const endCallback = request.onBodyChunkSent
+        ? () => request.onBodyChunkSent(chunk)
+        : undefined
```

Comment:
It's probably better and faster to just call `onBodyChunkSent` after/before write. Using the callback doesn't really do that much, it's no guarantee that it was sent, just that it's written to the kernels buffer.